### PR TITLE
Generic prow jobs for Framefrogs' component pool

### PR DESCRIPTION
**Pull request by rafalpotempa**
**Description**

Changes proposed in this pull request:

- Create generic jobs for following directories:
  - `components/application-connectivity-certs-setup-job`
  - `components/application-connectivity-validator`
  - `components/application-gateway`
  - `components/application-operator`
  - `components/application-registry`
  - `components/compass-runtime-agent`
  - `components/connector-service`
  - `components/connection-token-handler`
  - `components/connectivity-certs-controller`
  - `tests/compass-runtime-agent`
  - `tests/connector-service-tests`
  - `tests/application-gateway-tests`
  - `tests/application-operator-tests`
  - `tests/application-registry-tests`
  - `tests/application-connector-tests`
  - `tests/connection-token-handler-tests`

**Related issue(s)**

Applies to kyma-project/kyma/#5872

### CODE REVIEW

**Comment by rafalpotempa** [comment]
> Add tests

What do you mean?
These are present in config.yaml, e.g.:
```
- to: ../prow/jobs/kyma/tests/application-operator-tests/application-operator-tests-generic.yaml
        values:
          <<: *kyma_generic_component
          path: tests/application-operator-tests
          since: "1.7"
          optional: true
      - to: ../prow/jobs/kyma/tests/application-registry-tests/application-registry-tests-generic.yaml
        values:
          <<: *kyma_generic_component
          path: tests/application-registry-tests
          since: "1.7"
          optional: true
      - to: ../prow/jobs/kyma/tests/compass-runtime-agent/compass-runtime-agent-tests-generic.yaml
        values:
          <<: *kyma_generic_component
          path: tests/compass-runtime-agent
          since: "1.7"
          optional: true
      - to: ../prow/jobs/kyma/tests/connection-token-handler-tests/connection-token-handler-tests-generic.yaml
        values:
          <<: *kyma_generic_component
          path: tests/connection-token-handler-tests
          since: "1.7"
          optional: true
      - to: ../prow/jobs/kyma/tests/connector-service-tests/connector-service-tests-generic.yaml
        values:
          <<: *kyma_generic_component
          path: tests/connector-service-tests
          since: "1.7"
          optional: true
```

File: development/tools/jobs/kyma/tests_test.go [hunk 1]
```diff
@@ -21,15 +21,50 @@ var tests = []struct {
...
 			jobsuite.JobFileSuffix("tests"),
 		},
 	},
+	{path: "compass-runtime-agent", image: tester.ImageBootstrap20181204, suite: tester.NewGenericComponentSuite,
+		additionalOptions: []jobsuite.Option{
+			jobsuite.JobFileSuffix("tests-generic"),
+			jobsuite.Since(releases.Release17),
+			jobsuite.Optional(),
+		},
+	},
```
Comment (Szymongib) [inline development/tools/jobs/kyma/tests_test.go]: Didn't you miss those two `connection-token-handler-tests` and `connector-service-tests`?

File: templates/config.yaml [hunk 2]
```diff
@@ -122,6 +177,65 @@ templates:
...
+          <<: *kyma_generic_component
+          path: tests/connector-service-tests
+          since: "1.7"
+          optional: true
+      - to: ../prow/jobs/kyma/tests/console-backend-service/console-backend-service-tests-generic.yaml
+        values:
+          <<: *kyma_generic_component
+          path: tests/console-backend-service
+          since: "1.6"
+      - to: ../prow/jobs/kyma/tests/end-to-end/upgrade/upgrade-tests-generic.yaml
```
Comment (Szymongib) [inline templates/config.yaml]: I think this one is duplicated

File: templates/config.yaml [hunk 3]
```diff
@@ -122,6 +177,65 @@ templates:
...
+          path: tests/connection-token-handler-tests
+          since: "1.7"
+          optional: true
+      - to: ../prow/jobs/kyma/tests/connector-service-tests/connector-service-tests-generic.yaml
+        values:
+          <<: *kyma_generic_component
+          path: tests/connector-service-tests
+          since: "1.7"
+          optional: true
+      - to: ../prow/jobs/kyma/tests/console-backend-service/console-backend-service-tests-generic.yaml
```
Comment (Szymongib) [inline templates/config.yaml]: I think this one is duplicated

File: templates/config.yaml (hunk 2 above)
Comment (rafalpotempa) [reply templates/config.yaml]: Indeed. I removed the other and adjusted this one.

File: templates/config.yaml (hunk 3 above)
Comment (rafalpotempa) [reply templates/config.yaml]: Indeed. I removed the other and adjusted this one.

File: development/tools/jobs/kyma/tests_test.go (hunk 1 above)
Comment (rafalpotempa) [reply development/tools/jobs/kyma/tests_test.go]: True. I added those.

**Pull request merged**